week1/merge.py

```diff
@@ -7,10 +7,8 @@ def merge(line):
     Function that merges a single row or column in 2048.
     """
     line_without_zeros = [value for value in line if value != 0]
-    merged_line = try_merge(line_without_zeros)
-    while len(line) - len(merged_line) > 0: #still need to fix this guy
-        merged_line.append(0)
-    return merged_line;
+    return (try_merge(line_without_zeros) +
+           [0 for dummy_num in range(len(line) - len(line_without_zeros))])
```

Comment:
You took my advice. You went a _little_ farther than I would have taken it. For the sake of readability I would have left `merged_line` in there, something like this:
```python
non_zeros = [value for value in line if value != 0]
zeros = [0 for dummy_num in range(len(line) - len(non_zeros))]
return try_merge(non_zeros) + zeros
```

Reply:
:) It is possible I got a little carried away. I am running into something interesting when I try to implement your suggestion. It is not returning an array with 0, instead it is returning an empty array when I try to assign it to the variable zeros. Does this make sense to you? 
```
#this adds 0's to an array
merged_line = try_merge(non_zeros) + [0 for dummy_num in range(len(line) - len(non_zeros))]

#this appears to return an empty array in the program
zeros = [0 for dummy_num in range(len(line) - len(non_zeros))]
```

I might be doing something strange...